xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -2537,8 +2537,17 @@ void onError(Status error) {
         respTimer.cancel();
         respTimer = null;
       }
+
+      // Include node ID in RPC failure status messages that originate from XdsClient.
+      // We expect all watchers to propagate the description to the channel, and expose it
```

Comment:
I think this part of the comment is incorrect. We don't expect this will be propagated. We expect that _if it causes an error_ it will be propagated. I don't know why that is really even commented though, because that is the entire point of delivering an error.

I think I'd expect more:
> Include node ID in xds failures to allow cross-referencing with control plane logs when debugging.

Reply:
Thanks!

The first sentence is verbatim how the task is worded in the tracking spreadsheet :)
The point i was trying to make with these second sentence is that `ResourceWatcher` implementations do not erase the Status description, and propagate it to the user. F.e. it could've been erased here:

https://github.com/grpc/grpc-java/blob/e2449a7738e2d0e2ef324fd950c900d249685964/xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java#L249-L254

I checked that all current `ResourceWatcher`s do keep the status, but this may not always be the case. So I say we assume all of them do.

Reply:
I am fine assuming they all do. If they don't then they'd have a very poor error message that should be fixed independent of this.